### ISSUE TITLE
subprocess installed post-installation script returned error exit status 1

### ISSUE THREAD

**Issue by xbeta** [chef-cookbooks/mysql#335]
On Ubuntu Trusty (14.04)  using MySQL 5.5

```
Recipe: artifactory::mysql
  * mysql2_chef_gem[default] action install
    * mysql_client[default] action create
      * package[default :create mysql-client-5.5] action install (up to date)
      * package[default :create libmysqlclient-dev] action install (up to date)
       (up to date)
    * gem_package[mysql2] action install (up to date)
     (up to date)
  * mysql_service[artifactory] action create
    * package[artifactory :create mysql-server-5.5] action install

      ================================================================================
      Error executing action `install` on resource 'package[artifactory :create mysql-server-5.5]'
      ================================================================================

      Mixlib::ShellOut::ShellCommandFailed
      ------------------------------------
      Expected process to exit with [0], but received '100'
      ---- Begin output of apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 ----
      STDOUT: Reading package lists...
      Building dependency tree...
      Reading state information...
      The following extra packages will be installed:
        mysql-server-core-5.5
      Suggested packages:
        tinyca mailx
      The following NEW packages will be installed:
        mysql-server-5.5 mysql-server-core-5.5
      Preconfiguring packages ...
      0 upgraded, 2 newly installed, 0 to remove and 40 not upgraded.
      Need to get 0 B/5,314 kB of archives.
      After this operation, 52.6 MB of additional disk space will be used.
      Selecting previously unselected package mysql-server-core-5.5.
      (Reading database ... 91710 files and directories currently installed.)
      Preparing to unpack .../mysql-server-core-5.5_5.5.43-0ubuntu0.14.04.1_amd64.deb ...
      Unpacking mysql-server-core-5.5 (5.5.43-0ubuntu0.14.04.1) ...
      Selecting previously unselected package mysql-server-5.5.
      Preparing to unpack .../mysql-server-5.5_5.5.43-0ubuntu0.14.04.1_amd64.deb ...
      Unpacking mysql-server-5.5 (5.5.43-0ubuntu0.14.04.1) ...
      Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
      Processing triggers for ureadahead (0.100.0-16) ...
      Setting up mysql-server-core-5.5 (5.5.43-0ubuntu0.14.04.1) ...
      Setting up mysql-server-5.5 (5.5.43-0ubuntu0.14.04.1) ...
      Processing triggers for ureadahead (0.100.0-16) ...
      STDERR: start: Job failed to start
      invoke-rc.d: initscript mysql, action "start" failed.
      dpkg: error processing package mysql-server-5.5 (--configure):
       subprocess installed post-installation script returned error exit status 1
      Errors were encountered while processing:
       mysql-server-5.5
      E: Sub-process /usr/bin/dpkg returned an error code (1)
      ---- End output of apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 ----
      Ran apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 returned 100

      Resource Declaration:
      ---------------------
      # In /var/chef/cache/cookbooks/mysql/libraries/provider_mysql_service.rb

       32:         package "#{new_resource.name} :create #{server_package_name}" do
       33:           package_name server_package_name
       34:           version parsed_version if node['platform'] == 'smartos'
       35:           version new_resource.package_version
       36:           action new_resource.package_action
       37:         end
       38:

      Compiled Resource:
      ------------------
      # Declared in /var/chef/cache/cookbooks/mysql/libraries/provider_mysql_service.rb:32:in `block in <class:MysqlService>'

      package("artifactory :create mysql-server-5.5") do
        action [:install]
        retries 0
        retry_delay 2
        guard_interpreter :default
        package_name "mysql-server-5.5"
        version "5.5.43-0ubuntu0.14.04.1"
        timeout 900
        cookbook_name "artifactory"
      end


    ================================================================================
    Error executing action `create` on resource 'mysql_service[artifactory]'
    ================================================================================

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    package[artifactory :create mysql-server-5.5] (/var/chef/cache/cookbooks/mysql/libraries/provider_mysql_service.rb line 32) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '100'
    ---- Begin output of apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 ----
    STDOUT: Reading package lists...
    Building dependency tree...
    Reading state information...
    The following extra packages will be installed:
      mysql-server-core-5.5
    Suggested packages:
      tinyca mailx
    The following NEW packages will be installed:
      mysql-server-5.5 mysql-server-core-5.5
    Preconfiguring packages ...
    0 upgraded, 2 newly installed, 0 to remove and 40 not upgraded.
    Need to get 0 B/5,314 kB of archives.
    After this operation, 52.6 MB of additional disk space will be used.
    Selecting previously unselected package mysql-server-core-5.5.
    (Reading database ... 91710 files and directories currently installed.)
    Preparing to unpack .../mysql-server-core-5.5_5.5.43-0ubuntu0.14.04.1_amd64.deb ...
    Unpacking mysql-server-core-5.5 (5.5.43-0ubuntu0.14.04.1) ...
    Selecting previously unselected package mysql-server-5.5.
    Preparing to unpack .../mysql-server-5.5_5.5.43-0ubuntu0.14.04.1_amd64.deb ...
    Unpacking mysql-server-5.5 (5.5.43-0ubuntu0.14.04.1) ...
    Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
    Processing triggers for ureadahead (0.100.0-16) ...
    Setting up mysql-server-core-5.5 (5.5.43-0ubuntu0.14.04.1) ...
    Setting up mysql-server-5.5 (5.5.43-0ubuntu0.14.04.1) ...
    Processing triggers for ureadahead (0.100.0-16) ...
    STDERR: start: Job failed to start
    invoke-rc.d: initscript mysql, action "start" failed.
    dpkg: error processing package mysql-server-5.5 (--configure):
     subprocess installed post-installation script returned error exit status 1
    Errors were encountered while processing:
     mysql-server-5.5
    E: Sub-process /usr/bin/dpkg returned an error code (1)
    ---- End output of apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 ----
    Ran apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 returned 100

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/artifactory/recipes/mysql.rb

     32: mysql_service 'artifactory' do
     33:   version '5.5'
     34:   bind_address '0.0.0.0'
     35:   port '3306'
     36:   data_dir node['mysql']['data_dir']
     37:   initial_root_password mysql_creds['server_root_password']
     38:   action [:create, :start]
     39: end
     40:

    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/artifactory/recipes/mysql.rb:32:in `from_file'

    mysql_service("artifactory") do
      action [:create, :start]
      retries 0
      retry_delay 2
      guard_interpreter :default
      cookbook_name "artifactory"
      recipe_name "mysql"
      version "5.5"
      bind_address "0.0.0.0"
      port "3306"
      data_dir "/opt/mysql_data"
      initial_root_password "$mypassword!"
      package_action :install
      instance "artifactory"
      run_user "mysql"
      run_group "mysql"
    end


Running handlers:
[2015-05-05T00:51:20+00:00] ERROR: Running exception handlers
  - Chef::Handler::Datadog
Running handlers complete
[2015-05-05T00:51:22+00:00] ERROR: Exception handlers complete
[2015-05-05T00:51:22+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
Chef Client failed. 15 resources updated in 33.136009448 seconds
[2015-05-05T00:51:22+00:00] ERROR: mysql_service[artifactory] (artifactory::mysql line 32) had an error: Mixlib::ShellOut::ShellCommandFailed: package[artifactory :create mysql-server-5.5] (/var/chef/cache/cookbooks/mysql/libraries/provider_mysql_service.rb line 32) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '100'
---- Begin output of apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 ----
STDOUT: Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  mysql-server-core-5.5
Suggested packages:
  tinyca mailx
The following NEW packages will be installed:
  mysql-server-5.5 mysql-server-core-5.5
Preconfiguring packages ...
0 upgraded, 2 newly installed, 0 to remove and 40 not upgraded.
Need to get 0 B/5,314 kB of archives.
After this operation, 52.6 MB of additional disk space will be used.
Selecting previously unselected package mysql-server-core-5.5.
(Reading database ... 91710 files and directories currently installed.)
Preparing to unpack .../mysql-server-core-5.5_5.5.43-0ubuntu0.14.04.1_amd64.deb ...
Unpacking mysql-server-core-5.5 (5.5.43-0ubuntu0.14.04.1) ...
Selecting previously unselected package mysql-server-5.5.
Preparing to unpack .../mysql-server-5.5_5.5.43-0ubuntu0.14.04.1_amd64.deb ...
Unpacking mysql-server-5.5 (5.5.43-0ubuntu0.14.04.1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Processing triggers for ureadahead (0.100.0-16) ...
Setting up mysql-server-core-5.5 (5.5.43-0ubuntu0.14.04.1) ...
Setting up mysql-server-5.5 (5.5.43-0ubuntu0.14.04.1) ...
Processing triggers for ureadahead (0.100.0-16) ...
STDERR: start: Job failed to start
invoke-rc.d: initscript mysql, action "start" failed.
dpkg: error processing package mysql-server-5.5 (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 mysql-server-5.5
E: Sub-process /usr/bin/dpkg returned an error code (1)
---- End output of apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 ----
Ran apt-get -q -y install mysql-server-5.5=5.5.43-0ubuntu0.14.04.1 returned 100
[2015-05-05T00:51:22+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

**Comment by xbeta** [comment]
I somehow think I can fix it by moving it to 5.6 ?

**Comment by someara** [comment]
I can't reproduce this... 

Here is my test-kitchen output: https://gist.github.com/someara/62e1f0cd89d541a239ef

Oddly enough, #335 says 5.5 is broken but 5.6 works, 
Oddly enough, #331 says 5.6 is broken but 5.5 works.

I suspect you need to increase the ram in your VM.

Closing for now, please re-open if that doesn't fix it.
